demos/text_detection_demo/src/text_detection.cpp

```diff
@@ -191,49 +219,73 @@ std::vector<cv::RotatedRect> postProcess(const InferenceEngine::BlobMap &blobs,
-    new_cls_data_shape[1] = static_cast<int>(cls_shape[2]);
-    new_cls_data_shape[2] = static_cast<int>(cls_shape[3]);
-    new_cls_data_shape[3] = static_cast<int>(cls_shape[1]) / 2;
-
-    cv::Mat mask = decodeImageByJoin(cls_data, new_cls_data_shape, link_data, new_link_data_shape, cls_conf_threshold, link_conf_threshold);
-    std::vector<cv::RotatedRect> rects = maskToBoxes(mask, static_cast<float>(kMinArea),
-                                                     static_cast<float>(kMinHeight), image_size);
+    if (!kLocOutputName.empty() && !kClsOutputName.empty()) {
+//     PostProcessing for PixelLink Text Detection model
+        auto link_shape = blobs.at(kLocOutputName)->getTensorDesc().getDims();
+        size_t link_data_size = link_shape[0] * link_shape[1] * link_shape[2] * link_shape[3];
+        float *link_data_pointer =
+                blobs.at(kLocOutputName)->buffer().as<PrecisionTrait<Precision::FP32>::value_type *>();
+        std::vector<float> link_data(link_data_pointer, link_data_pointer + link_data_size);
+        link_data = transpose4d(link_data, ieSizeToVector(link_shape), {0, 2, 3, 1});
+        softmax(&link_data);
+        link_data = sliceAndGetSecondChannel(link_data);
+        std::vector<int> new_link_data_shape(4);
+        new_link_data_shape[0] = static_cast<int>(link_shape[0]);
+        new_link_data_shape[1] = static_cast<int>(link_shape[2]);
+        new_link_data_shape[2] = static_cast<int>(link_shape[3]);
+        new_link_data_shape[3] = static_cast<int>(link_shape[1]) / 2;
+
+        auto cls_shape = blobs.at(kClsOutputName)->getTensorDesc().getDims();
+        size_t cls_data_size = cls_shape[0] * cls_shape[1] * cls_shape[2] * cls_shape[3];
+        float *cls_data_pointer =
+                blobs.at(kClsOutputName)->buffer().as<PrecisionTrait<Precision::FP32>::value_type *>();
+        std::vector<float> cls_data(cls_data_pointer, cls_data_pointer + cls_data_size);
+        cls_data = transpose4d(cls_data, ieSizeToVector(cls_shape), {0, 2, 3, 1});
+        softmax(&cls_data);
+        cls_data = sliceAndGetSecondChannel(cls_data);
+        std::vector<int> new_cls_data_shape(4);
+        new_cls_data_shape[0] = static_cast<int>(cls_shape[0]);
+        new_cls_data_shape[1] = static_cast<int>(cls_shape[2]);
+        new_cls_data_shape[2] = static_cast<int>(cls_shape[3]);
+        new_cls_data_shape[3] = static_cast<int>(cls_shape[1]) / 2;
+
+        cv::Mat mask = decodeImageByJoin(cls_data, new_cls_data_shape,
+                       link_data, new_link_data_shape, cls_conf_threshold, link_conf_threshold);
+        rects = maskToBoxes(mask, static_cast<float>(kMinArea),
+                                                         static_cast<float>(kMinHeight), image_size);
+
+    }
+    else {
+        // PostProcessing for Horizontal Text Detection model
+        for (const auto& blob : blobs) {
+            if (blob.second->getTensorDesc().getDims()[0] == 100)
```

Comment:
I think this value may vary from model to model.

Reply:
You are right - it depends on the batch size. But to be honest, I don't even need to calculate kClsOutputName in the next [line](https://github.com/GalyaZalesskaya/open_model_zoo/blob/da991e8cb0b82c714adad571291725c779dead0a/demos/text_detection_demo/src/text_detection.cpp#L271), but I did it for sameness with PixelLink post-processing and to enable this [check](https://github.com/GalyaZalesskaya/open_model_zoo/blob/da991e8cb0b82c714adad571291725c779dead0a/demos/text_detection_demo/src/text_detection.cpp#L275) if passed text detection model is not supported. 
Probably I should remove or rewrite it